src/Control/Effect/Finally.hs

```diff
@@ -0,0 +1,18 @@
+module Control.Effect.Finally
```

Comment:
We add a new effect here -- `Finally` -- that has one operation:
```hs
onExit :: Has Finally sig m => m a -> m ()
```

This is used to clean up unpacked archive directories on program exit. We'd normally eagerly clean up these directories after discovering and running strategies, but we don't have a way to tell when the recursively-discovered strategies complete.

Reply:
Bummer.  Staggering those potentially large IO ops could be a lot more performant.  What's the exact reason we don't know when they're done?

Reply:
The discover actions don't keep track of the strategies they find -- they just fork the strategy as a new task to be run in parallel and forget about it. It's partially a consequence of how aggressively parallel we are